### PR TITLE
github action support

### PR DESCRIPTION
**Pull request by badhrinathpa**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - run: echo GIT_SHA_SHORT=$(git rev-parse --short HEAD) >> $GITHUB_ENV
+
+      - id: docker-login
+        uses: docker/login-action@v2.1.0
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKER_HUB_LOGIN }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Build and push "master-latest" Docker images
```
Comment (thakurajayL) [inline .github/workflows/docker.yml:39]: Are we using this image anywhere ? Else just compiling should be good enough ?

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        env:
+          DOCKER_TAG: master-${{ env.GIT_SHA_SHORT }
```
Comment (thakurajayL) [inline .github/workflows/docker.yml:41]: Do we get branch tag in the GitHub action ? As such this is ok for now, but want to know if we can use destination branch

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+concurrency:
+  group: ${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+  
+
+jobs:
+  build:
+    env:
+      DOCKER_REGISTRY: "docker.io/"
+      DOCKER_REPOSITORY: "badhrinathpa/"
```
Comment (thakurajayL) [inline .github/workflows/pull_request.yml:16]: Open item to check with ONF infra team ? @amolonf  would you be able to get this work with omec-project

File: .github/workflows/pull_request.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      # Format the code
+      - name: Go Format
+        run: |
+          make fmt
+
+      - name: Show all CI changes
+        run: |
+          git --no-pager diff
+
+      # Build again and commit
```
Comment (thakurajayL) [inline .github/workflows/pull_request.yml:35]: I see only build and no commit... Did I miss anything ?

File: Makefile [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Copyright 2019 free5GC.org
```
Comment (thakurajayL) [inline Makefile:1]: Keep... free5g copyright after SPDX header.

**Review by thakurajayL — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (badhrinathpa) [reply .github/workflows/docker.yml:39]: will comment out the docker push part. can enable later.

File: .github/workflows/docker.yml (hunk 2 above)
Comment (badhrinathpa) [reply .github/workflows/docker.yml:41]: Yes we can add branch name in tag :  
      - name: Build
        run: docker build -t udm/tag-example:${GITHUB_REF##*/} .

Currently we are running for master only. so didnt add branch name.

File: .github/workflows/pull_request.yml (hunk 4 above)
Comment (badhrinathpa) [reply .github/workflows/pull_request.yml:35]: good catch. will add the commit part.

**Review by thakurajayL — APPROVED** [review]

**Pull request merged**